api/v1/submissions/PKPSubmissionController.php

```diff
@@ -1691,4 +1708,151 @@ protected function getWriteDisabledErrors(string $schemaName, array $params): ar
 
         return $errors;
     }
+
+    /**
+     * Get JATS XML Files
+     */
+    public function getJats(Request $illuminateRequest): JsonResponse
+    {
+        $request = $this->getRequest();
+        $submission = $this->getAuthorizedContextObject(Application::ASSOC_TYPE_SUBMISSION);
+        
+        $publication = Repo::publication()->get((int) $illuminateRequest->route('publicationId'));
```

Comment:
I see you've got this checked manually below -- but if you use the 2nd parameter to `Repository::get` to specify the submission ID, then the auth policies for the submission will take care of the rest. That'll eliminate the need for extra checking, which is easily forgotten.

Reply:
I passed the `$submissionId` as a second parameter to the function.